lua/cfc_time/server/time_tracking.lua

```diff
@@ -140,19 +135,27 @@ function ctime:initPlayer( ply )
     end
 
     storage:PlayerInit( ply, now, function( data )
-        if not IsValid( ply ) then return end
-
         local isFirstVisit = data.isFirstVisit
         local sessionID = data.sessionID
 
+        local session = { joined = now }
         steamIDToPly[steamID] = ply
         ctime.sessionIDs[steamID] = sessionID
-        ctime.sessions[steamID] = { joined = now }
+        ctime.sessions[steamID] = session
+
+        if not IsValid( ply ) then
+            session.departed = now
+            return
+        end
 
         if isFirstVisit then return setupPly( 0, true ) end
 
         storage:GetTotalTime( steamID, function( total )
-            if not IsValid( ply ) then return end
+            if not IsValid( ply ) then
+                session.departed = now
+                return
+            end
+
             setupPly( total, false )
```

Comment:
Summary: If the player becomes invalid anywhere in here, we need to set their departed time so that the session closes out instead of dangling